### PR TITLE
[xwindow] part 4/N

### PR DESCRIPTION
**Pull request by Neumann-A**

### CODE REVIEW

**Review by JackBoosY — DISMISSED** [review]

File: ports/font-util/build.patch [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ #include <stdlib.h>
+ #include <string.h>
+-
++#if defined(_MSC_VER)
++# if defined(_M_X64)
++#  define SSIZE_MAX 9223372036854775807
++# else
++#  define SSIZE_MAX 2147483647
++# endif
++#endif
```
Comment (JavierMatosD) [inline ports/font-util/build.patch:16]: @Neumann-A would this work for arm64?
Would something like this work a bit better:
#if defined(_M_IX86) || defined(_M_ARM) #define the 32 bit one #else define the 64 bit one #endif

Comment (Neumann-A) [reply ports/font-util/build.patch:16]: or use `# if defined(_M_X64) || defined(_M_ARM64)` instead?

File: ports/font-util/build.patch [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+index 2b317c0b0..3ab84978f 100644
+--- a/bdftruncate.c	
++++ b/bdftruncate.c
+@@ -40,7 +40,13 @@
+ #include <stdio.h>
+ #include <stdlib.h>
+ #include <string.h>
+-
++#if defined(_MSC_VER)
++# if defined(_M_X64)
```
Comment (Neumann-A) [inline ports/font-util/build.patch:11]:
```suggestion
+# if defined(_M_X64) || defined(_M_ARM64)
```

**Comment by Neumann-A** [comment]
@JackBoosY remove the author:response tag.

**Comment by JackBoosY** [comment]
@JavierMatosD Request review again.

**Comment by Neumann-A** [comment]
@JackBoosY re-add reviewed label so that it gets re-reviewed.

File: ports/font-util/build.patch [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+--- a/bdftruncate.c	
++++ b/bdftruncate.c
+@@ -40,7 +40,13 @@
+ #include <stdio.h>
+ #include <stdlib.h>
+ #include <string.h>
+-
++#if defined(_MSC_VER)
++# if defined(_M_X64) || defined(_M_ARM64)
++#  define SSIZE_MAX 9223372036854775807
```
Comment (dan-shaw) [inline ports/font-util/build.patch:12]: Can we avoid hard coding this?

**Review by dan-shaw — DISMISSED** [review]

File: ports/font-util/build.patch (hunk 3 above)
Comment (Neumann-A) [reply ports/font-util/build.patch:12]: I mean I could include limits.h and use LLONG_MAX and LONG_MAX but does it really matter? SSIZE_MAX still needs to switch between those two depending on arch

Comment (JackBoosY) [reply ports/font-util/build.patch:12]: Generally we should avoid these magic numbers.

Comment (Neumann-A) [reply ports/font-util/build.patch]: ok i should use a magic macro instead ...... here you go

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**